delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/FileId.scala

```diff
@@ -0,0 +1,26 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.kernel.effect.migration.toCatsIOOps
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileId.iriExpander
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileRejection.InvalidFileId
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.ExpandIri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{IdSegment, IdSegmentRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.ProjectContext
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Tag.UserTag
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{ProjectRef, ResourceRef}
+import monix.bio.{IO => BIO}
+
+final case class FileId(id: IdSegmentRef, project: ProjectRef) {
+  def expandIri(fetchContext: ProjectRef => BIO[FileRejection, ProjectContext]): IO[(Iri, ProjectContext)] =
+    fetchContext(project).flatMap(pc => iriExpander(id.value, pc).map(iri => (iri, pc))).toCatsIO
```

Comment:
I don't see any calls to this?

Reply:
It's used all over `Files` to try to encapsulate that a bit - the diff is so big it's collapsed by default 😅